### DISCUSSION TITLE
Trigger on display connected vs disconnected - which option for external monitor?

### DISCUSSION THREAD

**Discussion by pinkgalaxy** [rxhanson/RectanglePro-Community#611] [Q&A]
Hey, @rxhanson  -

General question: 
Could you explain/give an example of when to use "trigger on display connected" vs "trigger on display disconnected"? Maybe even "trigger on wake," if it applies to my scenario. 

My scenario:
I use an external monitor with my Mac laptop that I connect via USB-C, and I want the app layouts to switch automatically when I plug/unplug the external monitor from my Macbook.

My question: 
Which options should I choose to trigger the layout changes for these 2 setups?

My setup: 
1. Setup 1: "o3" and "o2" = external monitor connected with Mac closed
2. Setup 2: "m3" and "m2" = external monitor disconnected from Mac; Mac is the main/only monitor

My current layouts:
DISPLAY CONNECTED:
![display connected](https://github.com/user-attachments/assets/9f127da1-1401-4704-b558-737de489e47e)

DISPLAY DISCONNECTED:
![display disconnected](https://github.com/user-attachments/assets/b9769395-0475-497a-b7f4-df8842d27003)

Thank you in advance!

I couldn't find a clear answer in these discussions:
- https://github.com/rxhanson/RectanglePro-Community/discussions/349
- https://github.com/rxhanson/RectanglePro-Community/discussions/361
- https://github.com/rxhanson/RectanglePro-Community/discussions/435
- https://github.com/rxhanson/RectanglePro-Community/discussions/522

**Comment by rxhanson** [comment]
It looks to me that you're selecting the proper options here, which leads me to think that perhaps your layouts are not getting applied as the display is connected/disconnected? If so, there is the possibility that the timing for the layout being applied vs when the display becomes reachable is not lining up. If this is the case, then perhaps we can troubleshoot further and get to the bottom of it.

**Comment by BrentEubanks** [comment]
I came here to ask essentially this same question - how do i correctly apply "trigger on display connected" vs "trigger on display disconnected" settings.

I was happy to find that someone had already asked and gotten an answer, but now I find myself confused by the answer because it seems backwards?

From the screenshots it appears that:

- O3 should activate when the external display is connected, but is set to "trigger on display disconnected" 
- M3 should activate when the external display is disconnected, but is set to "trigger on display connected" 

So, that seems backwards.
Can you check my understanding, or clarify?

Thanks!
